Quaver.Shared/Database/Maps/MapsetImporter.cs

```diff
@@ -243,6 +243,56 @@ public static void ImportFile(string path)
                 ConfigManager.AutoLoadOsuBeatmaps.Value = true;
                 NotificationManager.Show(NotificationLevel.Success, $"Successfully set the path for your .db file");
             }
+            // Folder with maps
+            else if (File.GetAttributes(path).HasFlag(FileAttributes.Directory))
+            {
+                var files = Directory.GetFiles(path);
+                var dirs = Directory.GetDirectories(path);
+
+                foreach (var subPath in files)
+                { 
+                    if (subPath.EndsWith(".qp") || subPath.EndsWith(".osz") || subPath.EndsWith(".sm") || subPath.EndsWith(".mcz") || subPath.EndsWith(".mc"))
+                    {
+                        NotificationManager.Show(NotificationLevel.Info, $"Scheduled {Path.GetFileName(subPath)} to be imported!");
+                        Queue.Add(subPath);
+                    }
+                }
+                foreach (var subDir in dirs)
+                {
+                    MapsetImporter.ImportFile(subDir);
+                }
+                // Took this from the first block
+                if (screen.Exiting)
```

Comment:
Maybe extract duplicate code to another function?

Reply:
I was really considering that instead of copying / pasting but wasn't sure if I should or not. It def makes more sense to move it to a function. I'll do it tomorrow and commit again.

Reply:
Okay I moved it to a new function and also added one more which simply checks if a file is of an accepted map type to avoid confusion later on if something were to change.